### DISCUSSION TITLE
BPM

### DISCUSSION THREAD

**Discussion by korekitty** [akira0245/MidiBard#44] [Ideas]
你好，我不会说中文。 [ [如何让MIDIBARD为不同乐曲自动切换音调和乐器?] ] 可以添加 BPM 吗？

Hello I can't speak Chinese language. [How to make MIDIBARD automatically switch key and instrument for different songs?] Can you add BPM?

**Comment by akira0245** [comment]
The BPM for a midi file is controlled by the file itself, and it may change in different sections of the piece, so it may not be wise to set a fixed BPM. 
Although MidiBard can adjust the overall playback speed of a piece of midi file, if you want precise BPM control, the best way is to edit the MIDI file directly.
